contracts/libs/gov/token-sale-proposal/TokenSaleProposalWhitelist.sol

```diff
@@ -90,17 +90,23 @@ library TokenSaleProposalWhitelist {
             .users[msg.sender]
             .purchaseInfo
             .lockedTokens;
+        EnumerableMap.AddressToUintMap storage requiredTokenLock = tier
+            .participationInfo
+            .requiredTokenLock;
+
+        (, uint256 lockedAmount) = lockedTokens.tryGet(tokenToUnlock);
+        (, uint256 requiredAmount) = requiredTokenLock.tryGet(tokenToUnlock);
+        uint256 overlock = lockedAmount < requiredAmount ? 0 : lockedAmount - requiredAmount;
 
         require(
             block.timestamp >= tier.tierInitParams.saleEndTime ||
-                !tier._checkUserLockedTokens(msg.sender),
+                !tier._checkUserLockedTokens(msg.sender) ||
```

Comment:
This may be flashloanable. 

1. The proposal for increasing the locked amount is accepted. 
2. The user implements a smart contract that executes a proposal.
3. Inside it 
  a. takes a flashloan
  b. locks tokens in TSP
  c. buys 
  d. executes the proposal
  e. unlocks tokens 
  f. repays the flashloan
  
Legit?